extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,86 @@
+import json
+from packageurl import PackageURL
+
+from conan.api.output import cli_out_write
+from conan.api.conan_api import ConanAPI
+from conan.cli.command import conan_command
+
+
+def licenses(id):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if id is None:
+        return []
+    else:
+        return [{"license": {
+            "id": id
```

Comment:
per definition of `license` is not guaranteed to be a valid SPDX license ID.
read https://docs.conan.io/2/reference/conanfile/attributes.html#license

per CycloneDX JSON schema definition, this is the list of supported SPDX licenses: 
https://cyclonedx.org/schema/spdx.schema.json


so assuming every value was a SPDX licen id might lead to invalid CycloneDX documents.


i'd recommend doing the following: 
1. store a set of well-known licenses in the code
2. if the given license is in well-known licenses,
   then: return as `id`-license  as in https://cyclonedx.org/docs/1.4/json/#tab-pane_components_items_licenses_items_license_oneOf_i0
   else: return as `name`-license as in https://cyclonedx.org/docs/1.4/json/#tab-pane_components_items_licenses_items_license_oneOf_i1

Reply:
I think the most simple approach is to just always use `name` as a starting point, this has the most value for open-source packages.

It's unfortunate the schema is a hardcoded list that does not include the license expressions https://spdx.github.io/spdx-spec/v2.3/SPDX-license-expressions/ which we recommend currently https://github.com/conan-io/conan-center-index/commit/4c0c17a4a0956d2c3d8b8610ceebeb7746b5301e

Reply:
Note that you can use license expressions, they are just not done under the "license" field, but under "expression":
https://cyclonedx.org/docs/1.4/json/#components_items_licenses_items_expression
e.g.
```
"licenses": [
  {
    "expression": "EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0"
  }
]
```
(https://github.com/CycloneDX/cyclonedx-dotnet-library/blob/main/tests/CycloneDX.Core.Tests/Json/v1.4/__snapshots__/SerializationTests.JsonRoundTripTest_valid-license-expression-1.4.json.snap)

Reply:
I switched to the cyclonedx python package which takes care of that

Reply:
> I switched to the cyclonedx python package which takes care of that

👍 
https://github.com/conan-io/conan-extensions/pull/66/files#r1286968212